File: Java/Problems/Arrays/InversionCount.java
package Problems.Arrays;

public class InversionCount {

    int getInversionCount(int[] array) {
        int count=0;

        for (int i=0; i<array.length; i++){
            for (int j = i+1; j<array.length; j++){
                if (array[i]>array[j]) count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        int[] a = new int[]{10,1,2,3,4};
        System.out.println(new InversionCount().getInversionCount(a));
    }
}

File: GO-Lang/Problems/Arrays/InversionCount/InversionCount.go
package main

import "fmt"

func GetInversionCount(A *[]int) int {
	count := 0

	for i, v := range *A {
		for j := i + 1; j < len(*A); j++ {
			if v > (*A)[j] {
				count++
			}
		}
	}
	return count
}

func main() {
	A := []int{10, 1, 2, 3, 4}
	fmt.Println("---->", GetInversionCount(&A))
}
